### PR TITLE
Mesh Mapping in Data Explorer

### PR DESCRIPTION
**Pull request by rahv**
This is mostly restructuring + usability changes.
I am in the process of decoupling mesh mapping and adding layers. As a first change, this adds a new dialog for mapping 2d meshes. As a tiny addition it is now possible to map all mesh nodes to a constant value.

### CODE REVIEW

**Comment by TomFischer** [comment]
Jenkins, test this please.

File: Applications/DataExplorer/DataView/MeshLayerEditDialog.cpp [hunk 1]
```diff
@@ -298,26 +281,11 @@ void MeshLayerEditDialog::accept()
...
-		new_mesh = new MeshLib::Mesh(*_msh);
-		const std::string imgPath ( this->_edits[0]->text().toStdString() );
-		const double noDataReplacementValue = this->_noDataReplacementEdit->text().toDouble();
-		if (!MeshLib::MeshLayerMapper::layerMapping(*new_mesh, imgPath, noDataReplacementValue))
-		{
-			delete new_mesh;
-			return;
-		}
-	}
+	MeshLib::Mesh* new_mesh (nullptr);
```
Comment (endJunction) [inline Applications/DataExplorer/DataView/MeshLayerEditDialog.cpp]: Could be a unique_ptr here.

File: Applications/DataExplorer/DataView/MeshMapping2DDialog.cpp [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+#include <QSettings>
+#include <QFileDialog>
+
+MeshMapping2DDialog::MeshMapping2DDialog(QDialog* parent)
+: QDialog(parent)
+{
+	setupUi(this);
+
+	StrictDoubleValidator* no_data_validator = new StrictDoubleValidator(this);
```
Comment (endJunction) [inline Applications/DataExplorer/DataView/MeshMapping2DDialog.cpp]: At this point I'm not sure who is going to delete the no_data_validator. The QLineEdit has empty dtor, so this looks like a memory leak to me.

File: Applications/DataExplorer/DataView/MeshLayerEditDialog.cpp [hunk 3]
```diff
@@ -298,26 +281,11 @@ void MeshLayerEditDialog::accept()
 	}
 
 	const unsigned nLayers = _layerEdit->text().toInt();
```
Comment (endJunction) [inline Applications/DataExplorer/DataView/MeshLayerEditDialog.cpp]: Unused variable `nLayers`

**Pull request merged**